app/helpers/patient_query_helper.rb

```diff
@@ -598,31 +615,50 @@ def advanced_filter_address(patients, filter)
     filter[:value].each do |field|
       case field[:name]
       when 'address-foreign'
-        patients = patients.where('patients.foreign_address_line_1 like ?', "%#{field[:value]&.downcase}%").or(
-          patients.where('patients.foreign_address_line_2 like ?', "%#{field[:value]&.downcase}%").or(
-            patients.where('patients.foreign_address_line_3 like ?', "%#{field[:value]&.downcase}%").or(
-              patients.where('patients.foreign_address_city like ?', "%#{field[:value]&.downcase}%").or(
-                patients.where('patients.foreign_address_zip like ?', "%#{field[:value]&.downcase}%").or(
-                  patients.where('patients.foreign_address_state like ?', "%#{field[:value]&.downcase}%").or(
-                    patients.where('patients.foreign_address_country like ?', "%#{field[:value]&.downcase}%")
-                  )
-                )
-              )
-            )
-          )
-        )
+        patients = if field[:value].present?
+                     patients.where('patients.foreign_address_line_1 like ?', "%#{field[:value]&.downcase}%").or(
+                       patients.where('patients.foreign_address_line_2 like ?', "%#{field[:value]&.downcase}%").or(
+                         patients.where('patients.foreign_address_line_3 like ?', "%#{field[:value]&.downcase}%").or(
+                           patients.where('patients.foreign_address_city like ?', "%#{field[:value]&.downcase}%").or(
+                             patients.where('patients.foreign_address_zip like ?', "%#{field[:value]&.downcase}%").or(
+                               patients.where('patients.foreign_address_state like ?', "%#{field[:value]&.downcase}%").or(
+                                 patients.where('patients.foreign_address_country like ?', "%#{field[:value]&.downcase}%")
+                               )
+                             )
+                           )
+                         )
+                       )
+                     )
+                   else
+                     patients.where(foreign_address_line_1: nil).or(patients.where(foreign_address_line_1: ''))
```

Comment:
is this meant to filter out all patients with any of these fields populated?  I'm seeing patients show up where one/two of the address fields are populated.  I would think that all the fields need to be blank in order to show up otherwise monitorees would potentially show up in both blank and non blank searches

Reply:
ignore this comment, I was searching for a blank domestic address but the monitoree had a foreign address defined